filter_plugins/cisco_nxos.py

```diff
@@ -0,0 +1,443 @@
+            continue
+
+        if line.startswith("vlan "):
+            if not found_vlanrange_line:
+                # Skip vlan range line
+                found_vlanrange_line = True
+                continue
+
+            # This is a VLAN config
+            object_name = int(line.removeprefix("vlan "))
+            if object_name in vlans:
+                if vlans[object_name].get("managed"):
+                    delete_section = False
+                    new_config.append(line)
+                    continue
+                else:
+                    delete_section = True
+                    continue
+            else:
+                delete_section = True
+                continue
+
+        if not delete_section:
+            new_config.append(line)
+
+    return new_config
+
+def generateVLANConfig(vlans):
+    config = []
+    for id,fields in vlans.items():
+        if fields.get("managed"):
+            continue
+
+        config.append("")  # Newline
+        config.append(f"vlan {str(id)}")
+        config.append(f"  name {fields['name']}")
+
+    return config
+
+def generateINTFConfig(interfaces):
+    """
+    This is the main method that generates the cisco NXOS config from the interface manifest
+    The output is ONLY the lines generated from the manifest, which needs to be combined with the existing running config
+
+    :param interfaces: The interfaces manifest
+    :type interfaces: dict
+    :return: The generated NXOS config
+    :rtype: list
+    """
+    def getLAGMembers(fields):
+        members = []
+        if "lag-members" in fields:
+            members += fields["lag-members"]
+
+        if "lacp-members-active" in fields:
+            members += fields["lacp-members-active"]
+
+        if "lacp-members-passive" in fields:
+            members += fields["lacp-members-passive"]
+
+        outlist = []
+        for member in members:
+            outlist.append(f"interface {member}")
+
+        return outlist
+
+    config_dict = {}
+
+    config = []
+    for intf,fields in interfaces.items():
+        intf_num = intf.split(" ")[-1]
+
+        # "fanout" field
+        if "fanout" in fields:
+            fanout_fields = fields["fanout"]
+            nxos_mode_str = ""
+            if fanout_fields["mode"] == "quad":
+                nxos_mode_str = "4x"
+            elif fanout_fields["mode"] == "dual":
+                nxos_mode_str = "2x"
+            elif fanout_fields["mode"] == "single":
+                nxos_mode_str = "1x"
+
+            intf_port = intf_num.split("/")[-1]
+            nxos_fanout_str = f"{fanout_fields['speed']}-{nxos_mode_str}"
+            config_dict[f"interface breakout module 1 port {intf_port} map {nxos_fanout_str}"] = []
+            continue
+
+        intf_cfg_label = intf
+        dict_key = f"interface {intf_cfg_label}"
+        if dict_key not in config_dict:
+            config_dict[dict_key] = []
+
+        is_portchannel = intf.startswith("port-channel")
```

Comment:
Shouldn't this be spelled `Port-channel`? In the config files, I see e.g.:

```
Port-channel 10:
  ...
Port-channel 11:
  ...
```

And I don't see anywhere we convert that to lower-case.

Reply:
Got it. I guess we don't have any cisco configs in `host_vars` at the moment.